source/fundamentals/data-formats/documents.txt

```diff
@@ -14,34 +14,351 @@ Documents
+     - ``java.lang.Boolean``
+   * - Date
+     - ``java.util.Date``
+   * - Document
+     - ``org.bson.Document``
+   * - Double
+     - ``java.lang.Double``
+   * - Int32
+     - ``java.lang.Integer``
+   * - Int64
+     - ``java.lang.Long``
+   * - Null
+     - ``null``
+   * - ObjectId``
+     - ``org.bson.types.ObjectId``
+   * - String
+     - ``java.lang.String``
+
+You can also customize the type mapping by specifying a custom codec.
+For more information on customizing mapped types, see our guide on using
+:doc:`Codecs </fundamentals/data-formats/codecs>`.
+
+In the following code snippet, we show how to instantiate and build a sample
+``Document`` instance that contains a variety of data types:
+
+.. code-block:: java
+   :copyable: true
+
+   Document author = new Document("_id", new ObjectId())
+       .append("name", "Gabriel García Márquez")
+       .append("lastUpdated", new Date(120, 2, 26))
+       .append("novels", Arrays.asList(
+           new Document("title", "One Hundred Years of Solitude").append("yearPublished", 1967),
+           new Document("title", "Chronicle of a Death Foretold").append("yearPublished", 1981),
+           new Document("title", "Love in the Time of Cholera").append("yearPublished", 1985)));
+
+You can insert this document into a collection using the following code:
+
+.. code-block:: java
+   :copyable: true
+
+  // MongoClient mongoClient = ...
+
+  MongoDatabase database = mongoClient.getDatabase("test_data");
+  MongoCollection<Document> collection = database.getCollection("authors");
+
+  InsertOneResult result = collection.insertOne(author);
+
+Once you perform a successful insert, you can retrieve the sample document
+data from the collection using the following code:
+
+.. code-block:: java
+   :copyable: true
+
+   import com.mongodb.client.model.Filters;
+   ...
+
+   Document doc = collection.find(Filters.eq("novels.yearPublished", 1967)).first();
+   if (doc != null) {
+       System.out.println("_id: " + doc.getObjectId("_id")
+           + ", name: " + doc.getString("name")
+           + ", lastUpdated: " + doc.getDate("lastUpdated"));
+
+       doc.getList("novels", Document.class).forEach((novel) -> {
+           System.out.println("title: " + novel.getString("title")
+               + ", yearPublished: " + novel.getInteger("yearPublished"));
+           });
+   }
+
+The output should resemble the following:
+
+.. code-block:: none
+   :copyable: false
+
+   _id: 5fb5fad05f734e3794741a35, name: Gabriel García Márquez, lastUpdated: Thu Mar 26 00:00:00 EDT 2020
+   title: One Hundred Years of Solitude, yearPublished: 1967
+   title: Chronicle of a Death Foretold, yearPublished: 1981
+   title: Love in the Time of Cholera, yearPublished: 1985
+
+For more information on retrieving and manipulating MongoDB data, see our
+:doc:`CRUD guide </fundamentals/crud>`.
+
+For more information on this class, see the
+:java-docs:`Document API documentation </apidocs/bson/org/bson/Document.html>`.
```

Comment:
Thanks for the suggestion. I think you mean `seealso`. I've been avoiding using it, but I think it doesn't really change the experience either way. I'll keep it in mind though since the change in formatting it produces could be a good signal to readers if used consistently.